dodiscover/constraint/fcialg.py

```diff
@@ -398,14 +398,135 @@ def _apply_rule4(
 
         return added_arrows, explored_nodes
 
+    def _apply_rule5(self, graph: EquivalenceClass, u: Column, a: Column) -> bool:
+        """Apply rule 5 of FCI algorithm.
+
+        For each A o-o U, if there is an uncovered (e.g. every triple is unshielded)
+        circle path p = <A, B, ..., D, U> such that A, D are not adjacent and U, B are
+        not adjacent, then orient A o-o U and every edge on p as undirected.
+
+        Note that unlike the other rules, R5 is a binary operator.
```

Comment:
```suggestion
        Note that unlike the other rules, R5 is a binary operator and operates on two nodes, rather than a triplet.
```

Just to be explicit.